test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -10,185 +10,246 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
 
 import games.strategy.triplea.delegate.Matches;
 import games.strategy.util.IntegerMap;
 
+@RunWith(MockitoJUnitRunner.class)
 public class UnitCollectionTest {
 
   @Mock
-  private final GameData gameData = Mockito.mock(GameData.class);
+  private GameData mockGameData;
+  private UnitType unitTypeOne;
+  private UnitType unitTypeTwo;
+  @Mock
+  private final PlayerID defaultPlayerID = Mockito.spy(new PlayerID("Default Player", true, false, null));
+  @Mock
+  private PlayerID otherPlayerID;
+  private UnitCollection unitCollection;
+
+  private Unit unitDefaultPlayer1;
+  private Unit unitDefaultPlayer2;
+  private Unit unitDefaultPlayer3;
+
+  private int unitCountDefaultPlayerUnitTypeOne;
+  private int unitCountDefaultPlayerUnitTypeTwo;
+
+  private int unitCountDefaultPlayer;
+  private int unitCountOtherPlayerUnitTypeOne;
+  private int unitCountOtherPlayerUnitTypeTwo;
+  private int unitCountOtherPlayer;
+
+  private int unitCountUnitTypeOne;
+  private int unitCountUnitTypeTwo;
+
+  private Unit unitOtherPlayer1;
+  private Unit unitOtherPlayer2;
+  private Unit unitOtherPlayer3;
+  private Unit unitOtherPlayer4;
+  private Unit unitOtherPlayer5;
+  private Unit unitOtherPlayer6;
+  private Unit unitOtherPlayer7;
+
+  int defaultPlayerNotifyChangedCounter = 0;
+
+  public final int getDefaultPlayerNotifyChangedCounter() {
+    return defaultPlayerNotifyChangedCounter;
+  }
 
   @Before
   public void setup() {

```

Comment:
This is a monster @Before method, but it is nicer to have this consolidated out of the @Test code. :+1: